dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java

```diff
@@ -409,6 +410,31 @@ private Organization getOrganizationByIdOptionalAuth(Optional<User> user, Long o
         }
     }
 
+    @DELETE
+    @Path("/{organizationId}/deleteNonApprovedOrg")
+    @Timed
+    @UnitOfWork
+    @ApiOperation(value = "Deletes an organization that has been rejected", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = void.class, hidden = true)
+    @Operation(operationId = "deleteRejectedOrPendingOrganization", summary = "Delete rejected organization", description = "Delete rejected organization")
+    public void deleteRejectedOrPendingOrganization(
+            @ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth Optional<User> user,
+            @Parameter(description = "Organization ID.", name = "organizationId", in = ParameterIn.PATH, required = true) @PathParam("organizationId") Long organizationId) {
+        Organization organization = organizationDAO.findById(organizationId);
+
```

Comment:
You need to check if user owns the org and/or is an admin or curator. Otherwise any user could delete any org. In fixing this, please add a test that ensures a user cannot delete an org they don't have permissions to.

Reply:
Reworked on the endpoint so that it now features additional privileges for admin/curator; they can now delete any organizations that is rejected/pending. Normal users can only delete their organizations if and only if they are MAINTAINERS of that organization. Other normal users cannot delete other organizations that they are not a part of.